pkg/generate/app/cmd/newapp_test.go

```diff
@@ -615,6 +651,22 @@ func TestRunAll(t *testing.T) {
 				imageStreams = append(imageStreams, tp)
 			case *deploy.DeploymentConfig:
 				got["deploymentConfig"] = append(got["deploymentConfig"], tp.Name)
+				if podTemplate := tp.Template.ControllerTemplate.Template; podTemplate != nil {
+					for _, volume := range podTemplate.Spec.Volumes {
+						var srcType string
+						if volume.VolumeSource.EmptyDir != nil {
+							srcType = "EmptyDir"
+						} else {
+							srcType = "UNKNOWN"
+						}
+						got["volumes"] = append(got["volumes"], volume.Name+"♥"+srcType)
```

Comment:
I still don't want to do it.

On Thu, Jun 11, 2015 at 5:16 PM, Martin Nagy <notifications@github.com>
wrote:

> In pkg/generate/app/cmd/newapp_test.go
> <https://github.com/openshift/origin/pull/3098#discussion_r32269241>:
>
> > @@ -615,6 +651,22 @@ func TestRunAll(t *testing.T) {
> >  				imageStreams = append(imageStreams, tp)
> >  			case *deploy.DeploymentConfig:
> >  				got["deploymentConfig"] = append(got["deploymentConfig"], tp.Name)
> > +				if podTemplate := tp.Template.ControllerTemplate.Template; podTemplate != nil {
> > +					for _, volume := range podTemplate.Spec.Volumes {
> > +						var srcType string
> > +						if volume.VolumeSource.EmptyDir != nil {
> > +							srcType = "EmptyDir"
> > +						} else {
> > +							srcType = "UNKNOWN"
> > +						}
> > +						got["volumes"] = append(got["volumes"], volume.Name+"♥"+srcType)
>
> Four tests in total. It would mean we would need to add another struct in
> there that would be set to nil in 15/16 cases. It would require changing
> other code as well (changing the type of variable got) and code that
> handles that. I think that encoding two strings in one is not such a bad
> solution, in comparison, especially since these are tests.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/3098/files#r32269241>.
>